Modules/Exercise05/Tests/Unit/Services/OrderServiceTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Modules\Exercise05\Tests\Unit\Services;
+
+use Modules\Exercise05\Services\OrderService;
+use Tests\SetupDatabaseTrait;
+use Tests\TestCase;
+
+class OrderServiceTest extends TestCase
+{
+    use SetupDatabaseTrait;
+
+    protected $orderService;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->orderService = new OrderService();
+    }
+
+    /**
+     * @dataProvider provideDetailOrder
+     */
+    public function test_it_handle_discount($infoBill, $detailOrder)
+    {
+        $result = $this->orderService->handleDiscount($detailOrder);
+
+        $this->assertEquals($result, $infoBill);
+    }
+
+    function provideDetailOrder()
```

Comment:
Trong provider này nên thêm có 1 case nữa check độ chính xác tới 2 chữ số thập phân vì cái round nó có thêm tham số thứ 2 là `2`
```
$infoBill['price'] = round(($detailOrder['price'] * 80) / 100, 2);
```
Cái này nice to have :smile: